File: HW/HW1/hw1.java
public class Solution {
    public static boolean isPalindrome(String input) {
        
        // Check if the string is empty
        if (input == "" || input == " ") {
            System.out.println("Invalid Input!");
            return false;
        }
    
        // Convert to lowercase and remove non-alphanumeric characters
        String newInput = input.toLowerCase().replaceAll("[^a-zA-Z0-9]", "");
    
        // Check if the cleaned string is a palindrome
        int left = 0;
        int right = newInput.length() - 1;
    
        while (left < right) {
            if (newInput.charAt(left) != newInput.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}

File: HW/HW1/hw1.py
def is_palindrome(input):
    # Check if the string is empty
    if input == "" or input == " ":
        print("Invalid Input!")
        return False
        
    # Convert to lowercase and remove non-alphanumeric characters
    newInput = ''.join(c.lower() for c in input if c.isalnum())

    # Check if the cleaned string is a palindrome
    left, right = 0, len(newInput) - 1
    while left < right:
        if newInput[left] != newInput[right]:
            return False
        left += 1
        right -= 1
    return True
